modules/gib.js

```diff
@@ -0,0 +1,145 @@
+const iconv = require('iconv-lite')
+const gametree = require('./gametree')
+const sgf = require('./sgf')
+const Board = require('./board')
+
+exports.parse = function (input) {
+
+    // Try UTF-8 encoding... in any case it's better than binary.
+
+    input = iconv.decode(Buffer.from(input, 'binary'), 'utf8')
+
+    let lines = input.split('\n')
+
+    let tree = gametree.new()
+    let root = {}
+    tree.nodes.push(root)
+
+    root.FF = ['4']
+    root.GM = ['1']
+    root.SZ = ['19']
+
+    let node = root
+
+    for (let n = 0; n < lines.length; n++) {
+
+        let line = lines[n].trim()
+
+        if (line.startsWith('\\[GAMEBLACKNAME=') && line.endsWith('\\]')) {
+
+            let s = line.slice(16, -2)
+            root.PB = [s]
+
+        } else if (line.startsWith('\\[GAMEWHITENAME=') && line.endsWith('\\]')) {
+
+            let s = line.slice(16, -2)
+            root.PW = [s]
+
+        } else if (line.startsWith('\\[GAMECONDITION=')) {
+
+            // Hard-coded the common komi cases.
+            // For better results, we could do a regex here instead.
+
```

Comment:
Yeah, this does seem a little restrictive... Since it seems like there's an obvious pattern to it. 😉